doc/tutorials/content/gpu_octree.rst

```diff
@@ -0,0 +1,235 @@
+        512.019 478.489 258.021 (squared distance 6316.54)
+        478.06 405.253 370.323 (squared distance 4214.79)
+        448.171 433.947 429.251 (squared distance 12411.8)
+        526.696 436.645 351.765 (squared distance 1382.18)
+        493.678 566.386 337.188 (squared distance 12850.2)
+        571.876 538.687 274.034 (squared distance 15908.6)
+    Radius search at (719.296 350.568 51.4605) within radius 81.2447. Found 3 results.
+        704.921 282.976 45.871 (squared distance 4806.6)
+        662.233 349.678 51.3719 (squared distance 3256.99)
+        780.983 333.923 48.8876 (squared distance 4088.97)
+    Radius search at (485.176 935.997 886.432) within radius 7.11413. Found 0 results.
+    Radius search at (23.8829 490.819 417.75) within radius 117.868. Found 3 results.
+        30.7867 442.83 328.752 (squared distance 10271.3)
+        54.8184 454.243 373.548 (squared distance 4248.56)
+        71.6224 502.673 441.761 (squared distance 2996.11)
+
+    Approximate nearest neighbor search at (60.1689 14.2143 637.145)
+        60.1326 76.0427 656.145 (squared distance 4183.76)
+    Approximate nearest neighbor search at (40.0743 226.868 413.113)
+        101.64 229.722 412.533 (squared distance 3798.78)
+    Approximate nearest neighbor search at (848.911 401.591 948.099)
+        885.188 428.836 963.89 (squared distance 2307.68)
+    Approximate nearest neighbor search at (664.014 665.193 931.986)
+        598.551 670.003 913.46 (squared distance 4651.77)
+    Approximate nearest neighbor search at (190.451 292.746 52.4367)
+        207.182 265.569 165.369 (squared distance 13772.3)
+    Approximate nearest neighbor search at (177.745 884.624 808.172)
+        232.255 834.861 807.517 (squared distance 5448.18)
+    Approximate nearest neighbor search at (499.687 453.289 332.382)
+        478.06 405.253 370.323 (squared distance 4214.79)
+    Approximate nearest neighbor search at (719.296 350.568 51.4605)
+        662.233 349.678 51.3719 (squared distance 3256.99)
+    Approximate nearest neighbor search at (485.176 935.997 886.432)
+        448.141 954.199 953.149 (squared distance 6154.03)
+    Approximate nearest neighbor search at (23.8829 490.819 417.75)
+        71.6224 502.673 441.761 (squared distance 2996.11)
+
+
+The following visualizations demonstrate the above three search methods in action. Note that some parameters are different from above, to highlight the functionality of the search methods.
+The first image depicts the initial cloud populated with random data.
+
+.. image:: images/gpu/octree/gpu_full.png
+  :width: 838
+
+Next, we can observe the results of K nearest search.
+
+.. image:: images/gpu/octree/gpu_knn.png
+  :width: 838
+
+Radius search results are shown below.
+
+.. image:: images/gpu/octree/gpu_radius.png
+  :width: 838
+
+The final image depicts results of approximate nearest search.
+
+.. image:: images/gpu/octree/gpu_approx.png
+  :width: 838
```

Comment:
Can you paint also the following:
- paint the query points in red
- For the approx NN method, in case it picks a suboptimal point, show the optimal point in green.